uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/waiting-approval-notification/components/waiting-approval-notification.component.js

```diff
@@ -0,0 +1,72 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'leave-absences/waiting-approval-notification/modules/components'
+], function (_, components) {
+  components.component('waitingApprovalNotification', {
+    templateUrl: ['settings', function (settings) {
+      return settings.pathTpl + 'components/waiting-approval-notification.html';
+    }],
+    controllerAs: 'waitingApprovalNotification',
+    controller: ['$log', '$q', '$rootScope', 'Session', 'OptionGroup', 'shared-settings', controller]
+  });
+
+  function controller ($log, $q, $rootScope, Session, OptionGroup, sharedSettings) {
+    $log.debug('Component: waiting-approval-notification');
+
+    var filters = {};
+    var leaveRequestStatuses = [];
+    var vm = this;
+    vm.eventName = 'updateStatus';
+
+    (function init () {
+      $q.all([
+        getManagerId(),
+        getStatusId()
+      ]).then(function () {
+        $rootScope.$emit('WaitingApproval:: Initialize Filters::' + vm.eventName, filters);
```

Comment:
instead of using events, just show `<leave-notification-badge />` when you have all the data that you need, or even better (i haven't tried myself), you could use the lifecycle hooks (https://docs.angularjs.org/guide/component) to detect changes in the bound data and use those changes to initialize `<leave-notification-badge />`